tests/system-tests/internal/sniffer/sniffer.go

```diff
@@ -0,0 +1,633 @@
+	}
+
+	snifferDeployment, err := createHostNetworkedPacketSnifferDeployment(
+		apiClient,
+		packetCapturePort,
+		packetCaptureProtocol,
+		packetCaptureInterface,
+		tcpdumpImage,
+		snifferDeploymentName,
+		snifferNamespace,
+		scheduleOnNodes,
+	)
+
+	if err != nil {
+		return snifferDeployment, err
+	}
+
+	retries := 12
+	pollInterval := 5
+
+	for i := 0; i < retries; i++ {
+		snifferDeployment, err = deployment.Pull(apiClient, snifferDeploymentName, snifferNamespace)
+		if err != nil {
+			return nil, err
+		}
+
+		// Check if NumberReady == DesiredNumberScheduled.
+		// In that case, simply return as all went well.
+		if snifferDeployment.Object.Status.ReadyReplicas == snifferDeployment.Object.Status.Replicas {
+			return snifferDeployment, nil
+		}
+		// If no port conflict error was found, simply sleep for pollInterval and then
+		// check again.
+		time.Sleep(time.Duration(pollInterval) * time.Second)
+	}
+
+	return snifferDeployment, fmt.Errorf("deployment still not ready after %d tries", retries)
+}
+
+// determineNetworkToolsImage will get the image and tag for network-tools from the release image.
+func determineNetworkToolsImage(apiClient *clients.Settings) (string, error) {
+	glog.V(100).Infof("Gathering the image and tag for tools")
+
+	imageName := "tools"
+	imageNamespace := "openshift"
+
+	imageStreamObj, err := imagestream.Pull(apiClient, imageName, imageNamespace)
+
+	if err != nil {
+		return "", fmt.Errorf("could not find imageStream for image %s in namespace %s; %w",
+			imageName, imageNamespace, err)
+	}
+
+	dockerImageString, err := imageStreamObj.GetDockerImage("latest")
+
+	if err != nil {
+		return "", fmt.Errorf("could not find dockerImage definition for the imageStream %s in namespace %s; %w",
+			imageName, imageNamespace, err)
+	}
+
+	return dockerImageString, nil
+}
+
+// createHostNetworkedPacketSnifferDeployment creates a host networked pod in namespace <namespace> on node <nodeName>.
+// It will start a packet sniffer, and it will log all GET request's source IP and the actual request string.
+//
+//nolint:funlen
+func createHostNetworkedPacketSnifferDeployment(
+	apiClient *clients.Settings,
+	packetCapturePort int,
+	packetCaptureProtocol,
+	packetCaptureInterface,
+	networkPacketSnifferImage,
+	snifferDeploymentName,
+	snifferNamespace string,
+	scheduleOnHosts []string) (*deployment.Builder, error) {
+	if packetCaptureProtocol != httpProtocol && packetCaptureProtocol != udpProtocol {
+		return nil, fmt.Errorf("createHostNetworkedPacketSnifferDeployment supports only %s and "+
+			"%s protocols, got: %s", httpProtocol, udpProtocol, packetCaptureProtocol)
+	}
+
+	glog.V(100).Infof("Creating the sniffer deployment with image %s", networkPacketSnifferImage)
+
+	cmd := tcpCaptureScript
+	if packetCaptureProtocol == udpProtocol {
+		cmd = udpCaptureScript
+	}
+
+	if strings.Contains(packetCaptureInterface, ".") {
+		packetCaptureInterface = strings.Split(packetCaptureInterface, ".")[0]
```

Comment:
in case of a vlan sub interface - why to listen on the _parent_ and not on the actual interface?